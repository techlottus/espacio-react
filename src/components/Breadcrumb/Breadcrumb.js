import React, { useEffect } from "react";

export const Breadcrumb = React.memo(({ data, onItem, onBack }) => {
  const breadcrumbRef = React.createRef();

  useEffect(() => {
    breadcrumbRef.current.data = {
      textItems: data.textItems || [],
      textColor: data.textColor || "brand",
      // iconColor: data.iconColor || '',
      // itemFocusColor: data.itemFocusColor || '',
      icon: data.icon || "",
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    breadcrumbRef.current.addEventListener("onItem", onItem);
    breadcrumbRef.current.addEventListener("onBack", onBack);
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-breadcrumb ref={breadcrumbRef}></lottus-breadcrumb>
    </>
  );
});
