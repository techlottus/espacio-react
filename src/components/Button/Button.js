

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


