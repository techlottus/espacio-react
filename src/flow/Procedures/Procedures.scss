@import "../../App.scss";

.procedures-container {
  width: 100%;
  @extend .headerapp-top;
}
