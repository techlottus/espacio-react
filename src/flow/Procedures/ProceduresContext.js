import { createContext } from "react";

export const ProceduresContext= createContext(null) 