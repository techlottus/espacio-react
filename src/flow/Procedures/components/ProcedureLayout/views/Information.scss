@import "../../../../../App.scss";

.info-view {
  font-family: "Nunito",sans-serif;
  padding: pixeltorem(38px) pixeltorem(35px) 0 pixeltorem(35px);

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    padding: pixeltorem(38px) pixeltorem(35px) 0 pixeltorem(35px);
  }

  .info-title {
    padding: pixeltorem(24px) 0;
    h2 {
      font-style: normal;
      font-weight: bold;
      font-size: pixeltorem(24px);
      line-height: pixeltorem(30px);
      @media only screen and (min-width: map-get($grid-breakpoints, m)) {
        font-size: pixeltorem(36px);
        line-height: pixeltorem(42px);
      }
    }
  }

  .info-text {
    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      padding-bottom: pixeltorem(24px);
    }
    p {
        font-style: normal;
        font-weight: normal;
        font-size: pixeltorem(14px);
        line-height: pixeltorem(19px);
      @media only screen and (min-width: map-get($grid-breakpoints, m)) {
        font-size: pixeltorem(16px);
        line-height: pixeltorem(22px);
      }
    }
  }

  .info-price {
      padding-bottom: pixeltorem(48px);
      padding-top: pixeltorem(24px);
      @media only screen and (min-width: map-get($grid-breakpoints, m)) {
        padding-top:0;
      }
    p {
        font-style: normal;
        font-weight: bold;        
        font-size: pixeltorem(24px);
        line-height: pixeltorem(30px);
    }
  }

  .info-btn {
    margin-top: pixeltorem(-24px);
    padding-bottom: pixeltorem(48px);
  }
}
