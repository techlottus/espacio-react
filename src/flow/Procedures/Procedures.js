import React, { useEffect, useState } from "react";
import HeaderApp from "../../shared/HeaderApp/HeaderApp";
import "./Procedures.scss";
import { ProceduresContext } from "./ProceduresContext";
import {
  stepsProcedures,
  typeflowsProcedures,
} from "../../constants/Procedures.constant";
import ProcedureLayout from "./components/ProcedureLayout/ProcedureLayout";
import { ModalProcedures } from "../../components/Modal/ModalProcedures/ModalProcedures";
import { useHistory } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { proceduresRequest } from "../../middlewares/proceduresMiddlewares/indexProceduresMiddleware";
import { getProceduresCostsService } from "../../middlewares/proceduresMiddlewares/proceduresCostsMiddleware";
import { ModalAccordion } from "../../components/Modal/ModalAccordion/ModalAccordion";
import { getProfileProceduresService } from "../../middlewares/proceduresMiddlewares/procedureMainMiddleware";
import { ModalExit } from "../../components/Modal/ModalExit/ModalExit";
import { setPathExitProcedures } from "../../actions/proceduresActions/proceduresAction";
import { unblockProcedures } from "../../helpers/unblock";
import { sendInfoTM } from "../../tagging/services/sendInfoTagManager";
import {
  modalProcedureBackEquivalenceStudiesTag,
  nextStepTwoEquivalenceStudiesTag,
} from "../../tagging/flows/procedures/equivalenceStudiesTag";
import {
  modalProcedureBackAcademicHistoryTag,
  modalProcedureNextAcademicHistoryTag,
} from "../../tagging/flows/procedures/academicHistoryTag";
import { certificateStudyVerifyTag } from "../../tagging/flows/procedures/certificateStudyTag";
import {
  modalVerifyBackStudyRecordTag,
  modalVerifyNextStudyRecordTag,
} from "../../tagging/flows/procedures/studyRecordTag";
import {
  modalVerifyBackReinstatementTag,
  modalVerifyNextReinstatementTag,
} from "../../tagging/flows/procedures/reinstatementTag";
import { ModalRequirementsProcedure } from "../../components/Modal/ModalRequirementsProcedure/ModalRequirementsProcedure";
import { ModalImage } from "../../components/Modal/ModalImage/ModalImage";

const Procedures = () => {
  const [steps, setSteps] = useState({
    ...stepsProcedures,
  });

  const history = useHistory();
  const [isModal, setIsModal] = useState({
    valid: false,
    info: null,
    file: null,
    type: null,
  });

  const [isModalExit, setIsModalExit] = useState({
    isShow: false,
    unblock: null,
  });

  const [isModalAcordeon, setIsModalAcordeon] = useState(false);
  const [isModalRequirements, setIsModalRequirements] = useState({
    isShow: false,
    info: []
  });
  const [isModalImage, setIsModalImage] = useState(false);
  const [isTabs,setIsTabs] = useState(true);
  const dispatch = useDispatch();
  const { proceduresAllOne,proceduresAllTwo } = useSelector((state) => state.procedures);
  const procedures = useSelector((state) => state.procedures);

  const handleStepsBody = (focus, titles ) => {
    setSteps((state) => {
      return {
        ...state,
        focus,
        titles
      };
    });
  };

  const handleSteps = (focus) => {
    setSteps((state) => {
      return {
        ...state,
        focus,
      };
    });
  };

  const handleAccordion = (state) => {
    setIsModalAcordeon(state);
  };

  const handleIsTabs = (isCheck) => {
    setIsTabs(isCheck);
  }

  const handleModalRequirements = (isShow,info) => {
    setIsModalRequirements({isShow,info});
  };

  const handleModalImage = (state) => {
    setIsModalImage(state);
  };

  const handleModalProcedures = (info) => {
    setIsModal(info);
  };
  const handleRequest = () => {
    setIsModal(false);
    dispatch(
      proceduresRequest(
        isModal.type,
        history,
        isModal.info,
        isModal.file,
        isModal.detailId
      )
    );
  };

  const [infoModal, setInfoModal] = useState({
    profile: null,
    information: null,
    file: null,
    title: null,
    cost: null,
  });
  const handleInfoModalProcedures = (info) => {
    setInfoModal(info);
  };
  useEffect(() => {
    let urlSplit = history.location.pathname.split("/");

    let procedureFlow = urlSplit[2];
    let unblock = history.block((location) => {
      if (unblockProcedures(location.pathname, procedureFlow)) {
        dispatch(setPathExitProcedures(location.pathname));
        setIsModalExit({
          isShow: true,
          unblock,
        });
        return false;
      }

      return true;
    });

    if (proceduresAllOne === null || proceduresAllTwo === null) {
      dispatch(getProfileProceduresService());
      dispatch(getProceduresCostsService());
    }
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <div className="procedures-container">
        <HeaderApp />
        <ProceduresContext.Provider
          value={{
            steps,
            handleStepsBody,
            handleSteps,
            isModal,
            isTabs,
            handleModalProcedures,
            handleInfoModalProcedures,
            handleAccordion,
            handleModalRequirements,
            handleIsTabs,
            handleModalImage
          }}
        >
          <ProcedureLayout />
          <ModalExit
            isShow={isModalExit.isShow}
            onClose={() => {
              setIsModalExit((state) => {
                return {
                  ...state,
                  isShow: false,
                };
              });
            }}
            onNext={() => {
              isModalExit.unblock();
              setIsModalExit({
                isShow: false,
                unblock: null,
              });
              setPathExitProcedures(null);
              history.push(procedures.pathExit);
            }}
            flow={"procedures"}
          />
        </ProceduresContext.Provider>
        <ModalProcedures
          isShow={isModal.valid}
          outside={() => setIsModal(false)}
          onClose={() => {
            setIsModal(false);
            switch (isModal.type) {
              case typeflowsProcedures.certificateStudy:
                sendInfoTM(window, certificateStudyVerifyTag, "back");
                break;
              case typeflowsProcedures.equivalenceStudy:
                sendInfoTM(window, modalProcedureBackEquivalenceStudiesTag);
                break;
              case typeflowsProcedures.academyHistory:
                sendInfoTM(window, modalProcedureBackAcademicHistoryTag);
                break;
              case typeflowsProcedures.studyRecord:
                sendInfoTM(window, modalVerifyBackStudyRecordTag, "back");
                break;
              case typeflowsProcedures.reinstatement:
                sendInfoTM(window, modalVerifyBackReinstatementTag, "back");
                break;
              default:
                break;
            }
          }}
          onNext={() => {
            handleRequest();
            switch (isModal.type) {
              case typeflowsProcedures.certificateStudy:
                sendInfoTM(window, certificateStudyVerifyTag, "next");
                break;
              case typeflowsProcedures.equivalenceStudy:
                sendInfoTM(window, nextStepTwoEquivalenceStudiesTag);
                break;
              case typeflowsProcedures.academyHistory:
                sendInfoTM(window, modalProcedureNextAcademicHistoryTag);
                break;
              case typeflowsProcedures.studyRecord:
                sendInfoTM(window, modalVerifyNextStudyRecordTag, "next");
                break;
              case typeflowsProcedures.reinstatement:
                sendInfoTM(window, modalVerifyNextReinstatementTag, "next");
                break;
              default:
                break;
            }
          }}
          infoModal={infoModal}
        />
        <ModalAccordion
          isShow={isModalAcordeon}
          onClose={() => {
            handleAccordion(false);
          }}
          outside={() => handleAccordion(false)}
        />
        <ModalRequirementsProcedure
          isShow={isModalRequirements.isShow}
          info={isModalRequirements.info}
          onClose={() => setIsModalRequirements({
            isShow: false,
            info: []
          })}
          outside={() => setIsModalRequirements({
            isShow: false,
            info: []
          })}
        />
        <ModalImage
          isShow={isModalImage}
          onClose={() => {
            handleModalImage(false);
          }}
          outside={() => handleModalImage(false)}
        />
      </div>
    </>
  );
};

export default Procedures;
