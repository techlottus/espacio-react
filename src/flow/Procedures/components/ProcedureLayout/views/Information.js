import React, { useContext, useEffect, useState } from "react";
import { useHistory } from "react-router";
import { Breadcrumb } from "../../../../../components/Breadcrumb/Breadcrumb";
import {
  breadcrumbInfo,
  btnInfoProcedures,
  btnInfoProceduresDownload,
  DownloadInformationActionableCatalog,
  feedbackInfoProcedures,
  FeedbackInformationActionableCatalog,
  typeflowsProcedures,
  typeRequestStatus,
} from "../../../../../constants/Procedures.constant";
import { Button } from "../../../../../components/Button/Button";
import "./Information.scss";
import {
  actionConfigProcedure,
  getTitleButtonInfo,
  getTitleButtonInfoDownload,
  getTitleInfoFeedback,
  setConfigInfoProcedures,
} from "../../../../../helpers/proceduresActions";
import { useSelector } from "react-redux";
import { ProceduresContext } from "../../../ProceduresContext";
import { downloadFile } from "../../../../../helpers/download";
import { notiObs } from "../../../../../observables/notificationObs";
import { typesNoti } from "../../../../../types/typeNoti";
import env from "../../../../../enviroment/environment";
import { sendInfoTM } from "../../../../../tagging/services/sendInfoTagManager";
import { requirementsEquivalenceStudiesTag } from "../../../../../tagging/flows/procedures/equivalenceStudiesTag";
import { certificateStudyRequiTag } from "../../../../../tagging/flows/procedures/certificateStudyTag";
import { downloadReinstatementTag } from "../../../../../tagging/flows/procedures/reinstatementTag";
import { Feedback } from "../../../../../components/Feedback/Feedback";
import { typesModalityBanner } from "../../../../../types/typesProcedures";
import { getValuesOfAuth } from "../../../../../helpers/auth";

const Information = ({ type }) => {
  const history = useHistory();

  const [textsInfo, setTexts] = useState({
    title: "",
    text: "",
    price: "",
  });

  const { handleAccordion, handleModalRequirements, handleModalImage } =
    useContext(ProceduresContext);

  const { stepsDegreeOnlineRequirement, stepsDegreeGraduateRequirement } =
    useSelector((state) => state.proceduresDegree);

  const [breadCrumb, setBreadCrumb] = useState({
    ...breadcrumbInfo,
  });

  const [modalRequirementsInfo, setModalRequirementsInfo] = useState({});

  const { proceduresAllOne, proceduresAllTwo } = useSelector(
    (state) => state.procedures
  );

  const [buttonInfo, setButtonInfo] = useState({
    ...btnInfoProcedures,
  });

  const [buttonInfoDownloadAdmision, setButtonInfoDownloadAdmision] = useState([
    {
      ...btnInfoProceduresDownload,
    },
  ]);

  const [buttonInfoDownload, setButtonInfoDownload] = useState({
    ...btnInfoProceduresDownload,
  });

  const [feedbackInfo, setFeedbackInfo] = useState({
    ...feedbackInfoProcedures,
  });

  const [configActions, setConfigAction] = useState([]);

  const { texts } = useSelector((state) => state.texts);

  const [feedbackText, setFeedbackText] = useState({});

  const [downloadDataAction, setDownloadDataAction] = useState({});

  const [flowType, setFlowType] = useState("");

  useEffect(() => {
    if (type.flow === typeflowsProcedures.admissionCertificate) {
      setButtonInfoDownloadAdmision(
        getTitleButtonInfoDownload(type.flow, texts).map((title) => {
          return {
            ...btnInfoProceduresDownload,
            title,
          };
        })
      );
    }
    setButtonInfo({
      ...btnInfoProcedures,
      title: getTitleButtonInfo(type.flow, texts),
    });
    setButtonInfoDownload({
      ...btnInfoProceduresDownload,
      title: getTitleButtonInfoDownload(type.flow, texts),
    });
    setFeedbackInfo({
      ...feedbackInfoProcedures,
      title: getTitleInfoFeedback(type.flow, texts),
    });
    setConfigInfoProcedures(
      type,
      setTexts,
      setBreadCrumb,
      breadCrumb,
      proceduresAllOne && proceduresAllTwo
        ? [...proceduresAllOne, ...proceduresAllTwo]
        : [],
      texts
    );
    setFlowType(type.flow);
    setConfigAction(actionConfigProcedure(type.flow));
    setFeedbackText(FeedbackInformationActionableCatalog[type.flow]);
    setDownloadDataAction(DownloadInformationActionableCatalog[type.flow]);
  }, [type.flow, proceduresAllOne, proceduresAllTwo]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    setModalRequirementsInfo({
      stepsDegreeGraduateRequirement,
      stepsDegreeOnlineRequirement,
    });
  }, [stepsDegreeOnlineRequirement, stepsDegreeGraduateRequirement]);

  const onItem = (e) => {
    switch (e.detail) {
      case "/":
        history.push(e.detail);
        break;
      case "/procedures-main":
        history.push(e.detail);
        break;
      default:
        break;
    }
  };

  const onBack = (e) => {
    history.push("/procedures-main");
  };

  const renderActions = (config) => {
    const allTypes = config.reduce((prev, curr, index) => {
      if (!prev.hasOwnProperty(curr)) {
        prev = { ...prev, [curr]: [index] };
        return prev;
      }
      prev = { ...prev, [curr]: [...prev[curr], index] };
      return prev;
    }, {});
    return config.map((type, index) => {
      const positionData = allTypes[type].reduce((prev, curr, i) => {
        if (curr === index) {
          return i;
        }
        return prev;
      }, 0);
      switch (type) {
        case "download":
          let buttonData = { ...buttonInfoDownload };
          if (flowType === typeflowsProcedures.admissionCertificate) {
            buttonData = { ...buttonInfoDownloadAdmision[positionData] };
          }
          const dataDownload = downloadDataAction[positionData];
          return (
            <div key={index} className="info-btn">
              <Button
                data={buttonData}
                onClick={() => {
                  switch (flowType) {
                    case typeflowsProcedures.scholarship:
                      return downloadFile(
                        "solicitud-beca.pdf",
                        getValuesOfAuth().urlsBecas?.urlSolicitud
                      )
                        .then(() => {
                          notiObs.next({
                            type: typesNoti.success,
                            text: texts.procedures
                              .downloadPDFSuccessReinstatement,
                          });
                        })
                        .catch(() => {
                          notiObs.next({
                            type: typesNoti.error,
                            text: texts.procedures
                              .downloadPDFErrorReinstatement,
                          });
                        });
                      break;
                  }
                  downloadFile(dataDownload.name, dataDownload.path)
                    .then(() => {
                      notiObs.next({
                        type: typesNoti.success,
                        text: dataDownload.successMsg,
                      });
                    })
                    .catch(() => {
                      notiObs.next({
                        type: typesNoti.error,
                        text: dataDownload.errorMsg,
                      });
                    });
                  sendInfoTM(window, downloadReinstatementTag);
                }}
              />
            </div> 
          );
          break;

        case "modal":
          return (
            <div key={index} className="info-btn">
              <Button
                data={buttonInfo}
                onClick={() => {
                  switch (flowType) {
                    case typeflowsProcedures.certificateStudy:
                      sendInfoTM(window, certificateStudyRequiTag);
                      return handleAccordion(true);
                      break;
                    case typeflowsProcedures.equivalenceStudy:
                      handleAccordion(true);
                      sendInfoTM(window, requirementsEquivalenceStudiesTag);
                      break;
                    case typeflowsProcedures.degreeInProgress:
                      setModalRequirementsInfo((state) => {
                        handleModalRequirements(
                          true,
                          state?.stepsDegreeOnlineRequirement
                        );
                        return { ...state };
                      });
                      break;
                    case typeflowsProcedures.degreeGraduate:
                      setModalRequirementsInfo((state) => {
                        handleModalRequirements(
                          true,
                          state?.stepsDegreeGraduateRequirement
                        );
                        return { ...state };
                      });
                      break;
                    case typeflowsProcedures.scholarship:
                      handleModalImage(true);
                      break;
                    default:
                      break;
                  }
                }}
              />
            </div>
          );
          break;

        case "feedback":
          const dataFeedback = feedbackText[positionData];
          return (
            <div key={index} className="info-btn">
              <Feedback data={feedbackInfo} text={dataFeedback?.description} />
            </div>
          );
          break;
        default:
          return null;
      }
    });
  };

  return (
    <>
      <div className="info-view">
        <div className="infoview-container">
          <div className="bread-info">
            <Breadcrumb data={breadCrumb} onItem={onItem} onBack={onBack} />
          </div>
          <div className="info-title">
            <h2>{textsInfo.title}</h2>
          </div>
          <div className="info-text">
            <p>{textsInfo.text}</p>
          </div>
          <div className="info-price">
            <p>{textsInfo.price}</p>
          </div>
          {renderActions(configActions)}
        </div>
      </div>
    </>
  );
};

export default Information;
