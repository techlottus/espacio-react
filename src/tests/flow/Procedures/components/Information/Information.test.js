import React from "react";
import { mount } from "enzyme";
import Information from "../../../../../flow/Procedures/components/ProcedureLayout/views/Information";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { act } from "react-dom/test-utils";
import { ProceduresContext } from "../../../../../flow/Procedures/ProceduresContext";
import { BrowserRouter } from "react-router-dom";
import { Breadcrumb } from "../../../../../components/Breadcrumb/Breadcrumb";
import { typesRequestErrors } from "../../../../../constants/error.constant";
import { Button } from "../../../../../components/Button/Button";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

window.scrollTo = jest.fn();

let initState = {
  procedures: {
    requestInquiryAll: null,
    proceduresAll: null,
    profile: null,
    errors: {
      [typesRequestErrors.getInformationProcedures]: {
        isError: false,
        msg: null,
      },
    },
  },
};

const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
const setIsModalAcordeon = jest.fn().mockImplementation(() => {});
let store = mockStore(initState);
const handleMount = (initState, type) => {
  store = mockStore(initState);
  return mount(
    <ProceduresContext.Provider value={{ setIsModalAcordeon }}>
      <Provider store={store}>
        <BrowserRouter>
          <Information type={type} history={historyMock} />
        </BrowserRouter>
      </Provider>
    </ProceduresContext.Provider>
  );
};

describe("Test <Information/>", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState, { flow: "current" });
    expect(wrapper).toMatchSnapshot();
  });

  test("should go back on", () => {
    const wrapper = handleMount(initState, { flow: "current" });
    act(() => {
      wrapper.find(Breadcrumb).props().onBack();
    });
    wrapper.update();
  });

  describe("Switch on btn with information", () => {
    test("Should switch between flows", () => {
      let wrapper = handleMount(initState, { flow: "certificate-study" });
      act(() => {
        wrapper.find(".info-btn").find(Button).props().onClick();
      });
      wrapper.update();
      wrapper = handleMount(initState, { flow: "equivalence-study" });
      act(() => {
        wrapper.find(".info-btn").find(Button).props().onClick();
      });
      wrapper.update();
      wrapper = handleMount(initState, { flow: "reinstatement" });
      act(() => {
        wrapper.find(".info-btn").find(Button).props().onClick();
      });
      wrapper.update();
    });
  });
  describe("Switch on breadcrum", () => {
    test("should redirect to dashboard", () => {
      const wrapper = handleMount(initState, { flow: "current" });
      act(() => {
        wrapper.find(Breadcrumb).props().onItem({ detail: "/" });
      });
      wrapper.update();
    });
    test("should redirect to main procedure", () => {
      const wrapper = handleMount(initState, { flow: "current" });
      act(() => {
        wrapper.find(Breadcrumb).props().onItem({ detail: "/procedures-main" });
      });
      wrapper.update();
    });
    test("should redirect to default", () => {
      const wrapper = handleMount(initState, { flow: "current" });
      act(() => {
        wrapper.find(Breadcrumb).props().onItem({ detail: "/main" });
      });
      wrapper.update();
    });
  });
});
