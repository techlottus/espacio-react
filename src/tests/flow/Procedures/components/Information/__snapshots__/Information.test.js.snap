// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Information/> should init component 1`] = `
<div
  className="info-view"
>
  <div
    className="infoview-container"
  >
    <div
      className="bread-info"
    >
      <lottus-breadcrumb />
    </div>
    <div
      className="info-title"
    >
      <h2 />
    </div>
    <div
      className="info-text"
    >
      <p />
    </div>
    <div
      className="info-price"
    >
      <p />
    </div>
  </div>
</div>
`;
